mockgen/model/model.go

```diff
@@ -452,3 +445,17 @@ func typeFromType(t reflect.Type) (Type, error) {
 	// TODO: Struct, UnsafePointer
 	return nil, fmt.Errorf("can't yet turn %v (%v) into a model.Type", t, t.Kind())
 }
+
+// impPath sanitizes the package path returned by `PkgPath` method of a reflect Type so that
+// it is importable. PkgPath might return a path that includes "vendor". These paths do not
+// compile, so we need to remove everything up to and including "/vendor/".
+// See https://github.com/golang/go/issues/12019.
+func impPath(imp string) string {
+	if strings.HasPrefix(imp, "vendor/") {
+		imp = "/" + imp
```

Comment:
return imp[7:]

Reply:
Is it possible that the path can be something like `vendor/foo/vendor/bar`? If that's possible, then we need to return `bar` instead of `foo/vendor/bar`, right?